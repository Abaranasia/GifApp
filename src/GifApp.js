import React, {useState} from 'react';
import AddCati from './components/AddCati';
//import CovidInfo from './components/CovidInfo';
import GifGrid from './components/GifGrid';
import Header from './components/Header';

export const GifApp = () => {

    //const categorias = ['tea', 'cat', 'dance']

    const [categorias, setcategorias] = useState(['cats']);

    /*
    const addCati = () => {
        setcategorias([...categorias, 'nueva']);
    }
    */
   
    return (
        <div>
            <Header/>
            <main>
                <AddCati setcategorias= {setcategorias}/>

                <ol>{
                    categorias.map (cati => (
                    // (<li key={cati}>{cati}</li>)
                    <GifGrid 
                            key={cati} 
                            categoria={cati} 
                    />
                    ))
                }
                </ol>
            </main> 
        </div>
    )
}