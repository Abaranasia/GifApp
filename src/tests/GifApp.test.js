
import React from "react";
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifApp } from '../GifApp';

describe('Pruebas de GifApp', () => {

    
    test('Debe mostrarse correctamente', () => {
        const wrapper= shallow(<GifApp />);

        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar una lista de categorias', () => {
        const categorias = ['cats', 'dogs'] //no podemos enviar datos al useState, por lo tanto creamos un array como alternativa que le pasamos al prop defaultCategories, justo para eso
        const wrapper= shallow(<GifApp defaultCategories={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })
    
/*     test('Debe llamar a AddCati con una categoria de prop:', () => {
    
        const cati="One Punch";
       // const wrapper= shallow(<AddCati categoria ={cati} />);
    })
    
    test('Debe llamar a GifGrid con una key y una categoria de props:', () => {
        
    }) */
    
});