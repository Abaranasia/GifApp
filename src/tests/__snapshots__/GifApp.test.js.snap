// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifApp Debe mostrar una lista de categorias 1`] = `
<div>
  <Header />
  <main>
    <AddCati
      setcategorias={[Function]}
    />
    <ol>
      <GifGrid
        categoria="cats"
        key="cats"
      />
      <GifGrid
        categoria="dogs"
        key="dogs"
      />
    </ol>
  </main>
</div>
`;

exports[`Pruebas de GifApp Debe mostrarse correctamente 1`] = `
<div>
  <Header />
  <main>
    <AddCati
      setcategorias={[Function]}
    />
    <ol>
      <GifGrid
        categoria="cats"
        key="cats"
      />
    </ol>
  </main>
</div>
`;
